### PR TITLE
168 cup 2024 fixes

### PR DESCRIPTION
**Pull request by gdoffe**
All fixes done during the 2024 French Robotic Cup.

### CODE REVIEW

File: platforms/pf-pami/motion_control.cpp [hunk 1]
```diff
@@ -126,7 +126,8 @@ static cogip::motion_control::PosePIDController linear_pose_controller(&linear_p
 static cogip::motion_control::SpeedFilterParameters linear_speed_filter_parameters(
     platform_min_speed_linear_mm_per_period,
     platform_max_speed_linear_mm_per_period,
-    platform_max_acc_linear_mm_per_period2
+    platform_max_acc_linear_mm_per_period2,
+    false
```
Comment (mlecriva) [inline platforms/pf-pami/motion_control.cpp:128]: It could be set defined as a constant with an explicit name.

**Review by mlecriva — CHANGES_REQUESTED** [review]
I don't have enough perspective to say more.

I think it's a good idea to merge the code related to moving between a robot and a pami to avoid having to duplicate changes and potential causes of error.

My comments are essentially remarks to take into account or not in anticipation of the transition from platform to object model.

File: platforms/pf-robot-actuators/PositionalActuator.cpp [hunk 2]
```diff
@@ -23,6 +23,12 @@ void PositionalActuator::actuate_timeout(int32_t command, uint32_t timeout_perio
     actuate(command);
 }
 
+void PositionalActuator::send_state(void) {
```
[the remark this answers is not in the document]
Comment (gdoffe) [reply platforms/pf-robot-actuators/PositionalActuator.cpp]: The idea now is not to send state regularly but only on events.

Could make sense to do it with a Visitor pattern as we could have that philosophy with uartpb (if it still have a future ^^)

File: platforms/pf-robot-actuators/include/Sensor.hpp [hunk 3]
```diff
@@ -0,0 +1,53 @@
+// Copyright (C) 2024 COGIP Robotics association <cogip35@gmail.com>
```
[the remark this answers is not in the document]
Comment (gdoffe) [reply platforms/pf-robot-actuators/include/Sensor.hpp:1]: Yes, all actuators classes will move outside platforms in dedicated libs and interfaces

File: lib/path/include/path/Pose.hpp [hunk 4]
```diff
@@ -80,7 +80,10 @@ class Pose : public cogip_defs::Pose {
             areDoublesEqual(O(), other.O()) &&
             areDoublesEqual(max_speed_ratio_linear(), other.max_speed_ratio_linear()) &&
             areDoublesEqual(max_speed_ratio_angular(), other.max_speed_ratio_angular()) &&
-            allow_reverse() == other.allow_reverse()
+            allow_reverse() == other.allow_reverse() &&
```
[the remark this answers is not in the document]
Comment (gdoffe) [reply lib/path/include/path/Pose.hpp:83]: This is used mainly when receiving a new pose order (also sometime called pose to reach or target pose ^^'). 

What we want to check is if we have to set a new pose order or not for the Engine, meaning we start a new move which implies things like resetting the PIDs.

So it is not just comparing two positions coordinates here, but also their associated properties.

But what is true, is that if it is not obvious here, then we have to rework it. ;)

File: platforms/pf-pami/motion_control.cpp (hunk 1 above)
Comment (gdoffe) [reply platforms/pf-pami/motion_control.cpp:128]: Same here, I deliberately did not port the modification as this platform will move to trash and the `pf-robot-motion-control` will be used as reference

File: platforms/pf-pami/motion_control.cpp [hunk 5]
```diff
@@ -408,9 +411,32 @@ void pf_start_motion_control(void)
...
+    // Reset pose straight filter state
+    pose_straight_filter.reset_current_state();
+}
+
 void pf_disable_motion_control()
 {
     pf_motion_control_platform_engine.disable();
+
+    // Small wait to ensure engine is disabled
+    ztimer_sleep(ZTIMER_MSEC, motion_control_thread_period_ms);
```
[the remark this answers is not in the document]
Comment (gdoffe) [reply platforms/pf-pami/motion_control.cpp:436]: I deliberately did not port the modification as this platform will move to trash and the `pf-robot-motion-control` will be used as reference

**Review by mlecriva — APPROVED** [review]
LGTM

**Pull request merged**